services/src/main/java/org/keycloak/crypto/ECDSASignatureProvider.java

```diff
@@ -7,7 +7,9 @@
 import org.bouncycastle.asn1.DERSequenceGenerator;
 import org.bouncycastle.asn1.x9.X9IntegerConverter;
 import org.keycloak.common.VerificationException;
+import org.keycloak.jose.jwk.JWK;
 import org.keycloak.models.KeycloakSession;
+import org.keycloak.util.JWKSUtils;
 
```

Comment:
How about using `ECDSAClientSignatureVerifierProvider` that is used for keycloak to verify the data that a client creates and signs (e.g. request object, JWS client assertion for client authentication)?

`ECDSASignatureProvider` is used for keycloak to sign and verify JWS (e.g. access token, ID token). DPoP Proof is the data that a client creates and signs and keycloak verify.